src/main/java/nextstep/courses/service/SessionService.java

```diff
@@ -0,0 +1,49 @@
+package nextstep.courses.service;
+
+import nextstep.courses.domain.Session;
+import nextstep.courses.domain.SessionRepository;
+import nextstep.users.domain.NsUser;
+import nextstep.users.domain.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class SessionService {
+    private final SessionRepository sessionRepository;
+    private final UserRepository userRepository;
+
+    public SessionService(SessionRepository sessionRepository, UserRepository userRepository) {
+        this.sessionRepository = sessionRepository;
+        this.userRepository = userRepository;
+    }
+
+    @Transactional
+    public Session save(Session session, Long courseId) {
+        return sessionRepository.save(session, courseId);
+    }
+
+    @Transactional(readOnly = true)
+    public Session findById(Long sessionId) {
+        return sessionRepository.findById(sessionId);
+    }
+
+    @Transactional(readOnly = true)
+    public List<Session> findByCourseId(Long courseId) {
+        return sessionRepository.findByCourseId(courseId);
+    }
+
+    @Transactional
+    public long enroll(Session session, NsUser nextStepUser) {
+        session.enroll(nextStepUser);
+        return sessionRepository.saveSessionUser(session, nextStepUser);
+    }
+
+    @Transactional(readOnly = true)
+    public List<NsUser> findAllUserBySessionId(Long sessionId) {
+        return sessionRepository.findAllUserBySessionId(sessionId);
```

Comment:
DM으로 이야기 나눈 뒤에 고민 해봤는데요! Join보다 아래 처럼 ids 를 이용해 조회 해오는게 더 확장성이 높을 것 같아요. 
User와 Session이 같은 DB를 사용하고 있다면, 지금 처럼 Join 으로 해도 되겠지만, 만약 다른 DB 또는 아예 다른 프로젝트일 경우에는 불가능 할 것 같아서요. 

지금 처럼 userRepository 를 이용해서 갖고오면 다른 DB 또는 다른 프로젝트 api 로 User를 갖고오기 용이할 것 같습니다 :) 

위 내용과 별개로 User가 Session의 하위 도메인이 아닌데 SessionRepository를 이용해서 User를 갖고오는것도 어색 한 것 같습니다. 🤔
```suggestion
        List<String> nextStepUserIds = sessionRepository.findAllUserBySessionId(sessionId);
        return userRepository.findByUserIds(nextStepUserIds);
```